Functions/PesterState.Tests.ps1

```diff
@@ -72,6 +72,36 @@ InModuleScope Pester {
         context "adding test result" {
             $p.EnterTestGroup('Describe', 'Describe')
 
+            it "times test accurately within 10 milliseconds" {
```

Comment:
The test placement is fine I think, but on a first look I am not sure what is being tested. Could you explain how that test works in a comment so we can refer to it when the test starts failing please?

Reply:
Definitely!

I hope you don't think I went overboard on the comments there but I added quite a few. I also discovered a misconception I had on the group level testing which contributed to it being so far off (+- 200 milliseconds) and now the test runs at +-10 milliseconds for the group as well as the individual test where it should be.

Let me know if any of my thinking for how the timings work is off but I think it's looking pretty stinking accurate on the timings now.

Thanks!